### ISSUE TITLE
fewer cell types predicted than in the reference

### ISSUE THREAD

**Issue by bsierieb1** [theislab/scarches#37]
hi, i'm following the [tutorial](https://scarches.readthedocs.io/en/latest/scanvi_surgery_pipeline.html) and i ran into the following issue:

my reference data set contains 9 cell types. when i run `vae.predict()`, only 5 are predicted but this is nothing out of ordinary because the remaining 4 contain few cells so no wonder they are not predicted correctly. the accuracy of the learned classifier is still 91%. importantly, the 5 that are predicted are not consecutive - they are ['0','1','2','3','8'], whereby '0' is represented by the largest number of cells.

when i run `model.predict()`, the only predicted IDs are ['1','2','3']. does it mean that only these reference cell types could be predicted or does it look like something went wrong?

my code:
```
sca.dataset.setup_anndata(source_adata,
                          batch_key='sample',
                          labels_key='leiden')

vae = sca.models.SCANVI(
    source_adata,
    "Unknown",
    n_layers=2,
    encode_covariates=True,
    deeply_inject_covariates=False,
    use_layer_norm="both",
    use_batch_norm="none",
)

vae.train(
    n_epochs_unsupervised=vae_epochs,
    n_epochs_semisupervised=scanvi_epochs,
    unsupervised_trainer_kwargs=dict(early_stopping_kwargs=early_stopping_kwargs),
    semisupervised_trainer_kwargs=dict(metrics_to_monitor=["elbo","accuracy"],
                                       early_stopping_kwargs=early_stopping_kwargs_scanvi),
    frequency=1
)

target_adata.obs['leiden'] = vae.unlabeled_category_

model = sca.models.SCANVI.load_query_data(
    target_adata,
    vae,
    freeze_dropout = True,
)
model._unlabeled_indices = np.arange(target_adata.n_obs)
model._labeled_indices = []

model.train(
    n_epochs_semisupervised=surgery_epochs,
    train_base_model=False,
    semisupervised_trainer_kwargs=dict(metrics_to_monitor=["accuracy", "elbo"],
                                       weight_decay=0,
                                       early_stopping_kwargs=early_stopping_kwargs_surgery
                                      ),
    frequency=1
)
```

**Comment by adamgayoso** [comment]
In the UMAP representation, do any cells from the target dataset mix with cells of label '0' from the source?

**Comment by bsierieb1** [comment]
assuming you're talking about the joint UMAP representation, cells from the reference and the query barely mix. and no, no query cells mix with '0', which i guess answers my question. does this mean that the reference is too different from the query? is there anything I can do in this situation?

**Comment by adamgayoso** [comment]
you might consider first trying a simpler option of using e.g., trVAE which has an extra MMD loss to aid mixing. If that ends up mixing well for you, then you can do label transfer via nearest neighbors.

But if it does not mix well, it may be a biological characteristic of your samples.

**Comment by bsierieb1** [comment]
thanks!

**Comment by bsierieb1** [comment]
@adamgayoso this may be a dumb question because i poorly understand the underlying algorithms but i will still ask. so, there was still no mixing between the query and the reference when i used trVAE. when i tried to transfer labels in Seurat earlier, i faced the same issue of not mixing in a joint reference-query representation. however, i was able to semi-successfully transfer labels because the TransferData function has an option to use the PCA of the _query data only_ for the weighting anchors. i realize that scArches works in a completely different way but is there a similar trick i can try?
